Marathon collector

matthewbentley opened the pull request:
Output from marathon is in the format:
```
{
  "version": "3.0.0",
  "gauges": {
    "api.mesosphere.marathon.core.event.impl.stream.HttpEventStreamActorMetrics.number-of-streams": {
      "value": 0
    },
    ...
  },
  "counters": {
    "GarbageCollector.totalGcs": {
      "count": 11652
    },
    ...
  },
  [other stuff]
}

dichiarafrancesco commented on src/fullerite/collector/marathon.go:
```diff
@@ -0,0 +1,243 @@
+
+	for k, v := range u {
+		f, exists := jsonToMetricMap[k]
+		if exists {
```
compact 197 - 198 ?

dichiarafrancesco replied:
same for all the if statements

dichiarafrancesco commented on src/fullerite/collector/marathon.go:
```diff
@@ -0,0 +1,243 @@
+				metrics, err = f(vmap, metrics)
+
+				if err != nil {
+					m.log.Warn("Could not decode ", err)
```
why we don't return nil, error here ? (same at line 209)

matthewbentley replied:
This way if there's an error parsing one of the metrics, it will keep going and still try to get the rest of them.

matthewbentley commented:
I addressed most of the comments, however the `for` loops with range and the `if` statements with `:=` and variables used outside the `if` block probably cannot be simplified

matthewbentley commented:
Line 160 prefixes all metrics with `marathon.` when building the `metric.Metric`
